http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -823,6 +786,142 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
       }
     } yield ()
 
+  private def deleteDiskResource(id: Either[Long,
+                                            WsmControlledResourceId
+                                 ], // can be either runtimeId or diskResourceId in order to get WSMcontrolled resource
+                                 workspaceId: WorkspaceId,
+                                 auth: Authorization
+  )(implicit
+    ev: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      diskResourceOpt <- id match {
```

Comment:
can this code be simplified by using the diskId?

Reply:
the issue is that in the RUNTIME_CONTROLLED_RESOURCE table only has runtimeId, wsmResourceId, and the resourceType, so to make this simpler it would involve more complicated querying, both to get the diskId from the runtimeId (multiple joins) before passing it into this func, and then to get the WsmControlledResource from the diskId (also multiple joins).